Sources/Holocron/FileSystemStorageProvider.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  FileSystemStorageProvider.swift
+//  Holocron-iOS
+//
+//  Created by Pranjal Satija on 4/15/19.
+//  Copyright © 2019 Bottle Rocket Studios. All rights reserved.
+//
+
+import Foundation
+
+// MARK: Base Struct
+public extension StorageProvider {
+    typealias FileSystem = FileSystemStorageProvider
+}
+
+public struct FileSystemStorageProvider {
```

Comment:
I chose to call it `FileSystemStorageProvider` instead of `FileSystem` because I think that's a more accurate name for what it is. The extension on `StorageProvider` that makes it available as `StorageProvider.FileSystem` sort of does what we're talking about. If you want, I could define a few public typealiases that make `Keychain` and `UserDefaults` available as `KeychainStorageProvider` and `UserDefaultsStorageProvider`?

Reply:
Makes sense. Let's not add the extra typealisases though.